platforms/nuttx/src/canbootloader/uavcan/main.c

```diff
@@ -1184,6 +1185,19 @@ __EXPORT int main(int argc, char *argv[])
 
 
 	} else {
+		if (bootloader.fw_image_descriptor->major_version > 0) {
```

Comment:
I think there's been too much 3rd party adoption outside of our control for this major_version check to be worth anything.